### PR TITLE
[IE CLDNN] Updated GPU device config

### PR DESCRIPTION
**Pull request by vladimir-paramuzov**
### Details:
 - Moved new plugin options and metrics to gpu/gpu_config.hpp header
 - Deprecated several existing options from cldnn_config and added corresponding new ones in gpu_config header

### CODE REVIEW

File: inference-engine/include/cldnn/cldnn_config.hpp [hunk 1]
```diff
@@ -159,6 +104,7 @@ DECLARE_CONFIG_KEY(DUMP_KERNELS);
  *
  * For values TUNING_CREATE and TUNING_RETUNE the file will be created if it does not exist.
  */
+INFERENCE_ENGINE_DEPRECATED("The config key will be removed")
```
Comment (ilya-lavrenov) [inline inference-engine/include/cldnn/cldnn_config.hpp:107]: should not we deprecated the keys below as well?

File: docs/snippets/GPU_Kernels_Tuning.cpp [hunk 2]
```diff
@@ -4,9 +4,11 @@
 int main() {
 using namespace InferenceEngine;
 //! [part0]
-Core ie;          
+Core ie;
+IE_SUPPRESS_DEPRECATED_START
   ie.SetConfig({{ CONFIG_KEY(TUNING_MODE), CONFIG_VALUE(TUNING_CREATE) }}, "GPU");
   ie.SetConfig({{ CONFIG_KEY(TUNING_FILE), "/path/to/tuning/file.json" }}, "GPU");
+IE_SUPPRESS_DEPRECATED_END
```
Comment (ilya-lavrenov) [inline docs/snippets/GPU_Kernels_Tuning.cpp:11]: I think snippets should demonstrate new API only. If we deprecated this option, maybe better to remove its demonstration to users.

File: inference-engine/include/cldnn/cldnn_config.hpp [hunk 3]
```diff
@@ -11,47 +11,11 @@
 #pragma once
 
 #include "ie_plugin_config.hpp"
+#include "ie_api.h"
+#include "gpu/gpu_config.hpp"
```
Comment (ilya-lavrenov) [inline inference-engine/include/cldnn/cldnn_config.hpp:15]: maybe we need to remove `gpu/gpu_config.hpp` include from here? In this case users *have* to migrate to new config file to utilize its options

Comment (vladimir-paramuzov) [reply inference-engine/include/cldnn/cldnn_config.hpp:15]: I have moved GPU_THROUGHPUT_STREAMS option to gpu_config.hpp, so for backward compatibility it have to be included into cldnn_config.hpp

File: inference-engine/include/cldnn/cldnn_config.hpp (hunk 1 above)
Comment (vladimir-paramuzov) [reply inference-engine/include/cldnn/cldnn_config.hpp:107]: Fixed

File: docs/snippets/GPU_Kernels_Tuning.cpp (hunk 2 above)
Comment (vladimir-paramuzov) [reply docs/snippets/GPU_Kernels_Tuning.cpp]: This snipped was removed

**Review by ilya-lavrenov — APPROVED** [review]

**Review by lznamens — APPROVED** [review]

**Review by andrew-zaytsev — APPROVED** [review]

**Pull request merged**